### DISCUSSION TITLE
How to customise the URL in the online editor?

### DISCUSSION THREAD

**Discussion by schulle4u** [datenstrom/community#326] [Ask a question] [answered]
Hello, 

there are already some page title customizations, but can it be made possible to also customize the `@title` url slug for the blog plugin when creating pages through the editor? As an example: 

```
---
Title: This is a blog page with a very very very very long title
TitleSlug: custom-url-title
Published: 2018-05-27 12:00:00
Template: blog
---
```

Currently you can achive this by using `Title` as url title slug and `TitleHeader` / `TitleContent` to display the actual titles. This however requires you to modify some content snippets to also display the right titles in page listings. Maybe I'm missing something here?

Steffen

**Comment by schulle4u** [comment]
I was able to patch the edit plugin, seems this is only one line of additional code in the `getPageNewLocation()` function. 

```
$location = preg_replace("/@slug/i", $this->getPageNewData($rawData, "titleSlug"), $location);
```

Then I can set `BlogNewLocation: @slug` in my config.ini and work with my metadata as described in the first post. Would be great to have this included in the official plugin, assuming this doesn't break anything else.

Steffen

**Comment by markseuffert** [comment]
I would like to understand what you want to do, Steffen. Can you explain a bit more please. What problem do you want to solve? Renaming the content file in the file system is not a good solution? Do you have some real life examples where title and URL are different?

**Comment by schulle4u** [comment]
I'm currently migrating a audio/podcast website into yellow's blog plugin. The episode URLs should have a short, handy identifier to make them almost identical to the media files. So if the media file of episode 1 is named /media/downloads/epi001.mp3, the url to the belonging blog article should look something similar, e.g. /blog/epi001 while the article title can be a little more descriptive (episode 1: introduction). Mayby this is rather a personal taste, but I like to unify such things. I could rename the txt files after posting the articles, but thought it is a good idea to have this feature already available in the editor. 

Generally I don't like endless URLs that are automatically generated from article titles. Yellow's edit plugin furthermore doesn't transliterate some characters such as umlauts, which may have a negative effect on some (older/incompatible) systems. Most blog systems I know of have an option to customize the URL slug. But if this is not possible or pointless for Yellow, I'll deal with it. :)

Hope my explanation makes sense to someone. 

Steffen

**Comment by markseuffert** [comment, marked as answer]
Thank you for the detailed explanation. 

I did run into the same problems in the past and also don't like it when you have to adjust your way of working to the tool you're using. I took your suggestion and implemented it in a slightly different way. It gives you the possibility to override the title used for saving a page. You don't need to change your blog configuration, just add `TitleSlug` to the top of your page.  

For now it's an experimental feature. Let me know how this works for you.

**Comment by schulle4u** [comment]
Slightly different, but better than before. Now you even don't have to enter a TitleSlug if you're satisfied with the auto-generated title. And customizing content configuration directly in the pages is always a better solution than messing around with system configuration. Many thanks, works perfectly for me.

**Comment by markseuffert** [comment]
Good to hear. I suggest to use it a couple of weeks, before we add it to the official documentation.

**Comment by markseuffert** [comment]
Added to documentation.
